core/src/main/java/hudson/model/User.java

```diff
@@ -1049,24 +1051,17 @@ public String resolveCanonicalId(String idOrFullName, Map<String, ?> context) {
                 return existing.getId();
             }
             if (SECURITY_243_FULL_DEFENSE) {
-                Jenkins j = Jenkins.getInstance();
-                if (j != null) {
-                    if (!resolving.get()) {
-                        resolving.set(true);
-                        try {
-                            UserDetails userDetails = j.getSecurityRealm().loadUserByUsername(idOrFullName);
-                            if (userDetails == null) {
-                                throw new NullPointerException("hudson.security.SecurityRealm should never return null. "
-                                        + j.getSecurityRealm() + " returned null for idOrFullName='" + idOrFullName + "'");
-                            }
-                            return userDetails.getUsername();
-                        } catch (UsernameNotFoundException x) {
-                            LOGGER.log(Level.FINE, "not sure whether " + idOrFullName + " is a valid username or not", x);
-                        } catch (DataAccessException x) {
-                            LOGGER.log(Level.FINE, "could not look up " + idOrFullName, x);
-                        } finally {
-                            resolving.set(false);
-                        }
+                if (!resolving.get()) {
+                    resolving.set(true);
+                    try {
+                        UserDetails userDetails = UserDetailsCache.get().loadUserByUsername(idOrFullName);
```

Comment:
so this looks like you are only caching that the user exists...
you still pay the penalty if the user does not exist and you have a large user list to trawl through on the authentication side.  (which would mean those Users used to display the Changelog would not be found and not cached either).